extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,241 @@
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryGetBindingDataContent(ModelBindingData bindingData, out IDictionary<string, string> bindingDataContent)
+        {
+            bindingDataContent = bindingData.ContentType switch
+            {
+                Constants.JsonContentType => new Dictionary<string, string>(bindingData.Content.ToObjectFromJson<Dictionary<string, string>>(), StringComparer.OrdinalIgnoreCase),
+                _ => null
+            };
+
+            return bindingDataContent is not null;
+        }
+
+        private async Task<object?> ConvertModelBindingDataAsync(IDictionary<string, string> content, Type targetType, ModelBindingData bindingData)
+        {
+            content.TryGetValue(Constants.Connection, out var connectionName);
+            content.TryGetValue(Constants.ContainerName, out var containerName);
+            content.TryGetValue(Constants.BlobName, out var blobName);
+
+            if (string.IsNullOrEmpty(connectionName) || string.IsNullOrEmpty(containerName))
+            {
+                throw new ArgumentNullException("'Connection' or 'ContainerName' cannot be null or empty");
+            }
+
+            var result = await ToTargetType(targetType, connectionName, containerName, blobName);
+            return result;
+        }
+
+        private async Task<object?> ToTargetType(Type targetType, string connectionName, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String) => await GetBlobString(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(Stream) => await GetBlobStream(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(Byte[]) => await GetBlobBinaryData(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobBaseClient) => CreateBlobClient<BlobBaseClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient) => CreateBlobClient<BlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient) => CreateBlobClient<BlockBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient) => CreateBlobClient<PageBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient) => CreateBlobClient<AppendBlobClient>(connectionName, containerName, blobName),
+            Type _ when targetType == typeof(BlobContainerClient) => CreateBlobContainerClient(connectionName, containerName),
+            _ => await DeserializeToTargetObject(targetType, connectionName, containerName, blobName)
+        };
+
+        private IEnumerable<object> ToTargetTypeCollection(Type targetType, IEnumerable<object> blobCollection) => targetType switch
```

Comment:
Wondering if this can be simplified the same way we're using generics for `CreateBlobClient<T>`